sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/TargetScaleThrottler.cs

```diff
@@ -0,0 +1,147 @@
+                lock (lockObject)
+                {
+                    // Removing expired items from history. We want to keep the history item around the threshold so adding 2 secs.
+                    var threshold = executionTime - TimeSpan.FromSeconds(ExpirationThresholdInseconds + 2);
+                    bool historyCleared = false;
+                    for (int i = _history.Count - 1; i >= 0; i--)
+                    {
+                        if (_history[i].Metrics.Timestamp < threshold)
+                        {
+                            historyCleared = true;
+                            _history.RemoveAt(i);
+                        }
+                    }
+
+                    // Adding new history item
+                    EventHubTragetScalerReult currentResult = new EventHubTragetScalerReult
+                    {
+                        TargetScalerResult = result,
+                        Metrics = metric,
+                        ChangeWorkerCount = result.TargetWorkerCount - _lastTargetScalerResult.TargetWorkerCount
+                    };
+                    _history.Add(currentResult);
+                    int changeWorkerCount = currentResult.ChangeWorkerCount;
+
+                    // Throttle scale down based on latest scale up
+                    if (changeWorkerCount < 0 && (executionTime - _lastScaleUpTime).TotalSeconds < ThrottleScaleDownIntervalInSeconds)
+                    {
+                        _logger.LogInformation($"Throttling scale down, since last scale up time was within {ThrottleScaleDownIntervalInSeconds} seconds. (LastScaleUpTime: '{_lastScaleUpTime}', LastTargetWorkerRequest: '{_lastTargetScalerResult.TargetWorkerCount}')");
+                        return _lastTargetScalerResult;
+                    }
+
+                    // If there we have enough history try to throttle based on history
+                    if ((historyCleared) && (Math.Abs(changeWorkerCount) > 0))
+                    {
+                        long totalProcessedMessages = 0;
+                        long totalBacklog = 0;
+                        var samplesCount = _history.Count - 1;
+                        for (int i = samplesCount; i > 0; i--)
+                        {
+                            totalProcessedMessages += GetProcessedEventsCount(_history[i - 1].Metrics, _history[i].Metrics);
+                            totalBacklog += _history[i].Metrics.EventCount;
+                        }
+
+                        // Throttle scale up based average backlog and execution rate
+                        if (changeWorkerCount > 0 && totalBacklog < (double)totalProcessedMessages / ScaleUpFactor)
+                        {
+                            _logger.LogInformation($"Throttling scale up due average backlog is less then '{ScaleUpFactor}%' of total processed messages. (TotalProcessedMessages: '{totalProcessedMessages}', TotalBackLog: '{totalBacklog}', SamplesCount: '{samplesCount}'");
```

Comment:
```suggestion
                            _logger.LogInformation($"Throttling scale up because average backlog is less than '{ScaleUpFactor}%' of total processed messages. (TotalProcessedMessages: '{totalProcessedMessages}', TotalBackLog: '{totalBacklog}', SamplesCount: '{samplesCount}'");
```